### ISSUE TITLE
require-computed-property-dependencies doesn't recognize services

### ISSUE THREAD

**Issue by BlueRaja** [ember-cli/eslint-plugin-ember#477]
There's no reason to include services in the dependency list, but this flag requires them.

Example:
```
import { inject as service } from '@ember/service';

export default EmberObject.extend({
  myService: service(),
  myComputed: computed(function () {
    this.get('myService').someFunction(); // Incorrectly flagged as missing dependency
```

**Comment by bmish** [comment]
My understanding is that you should have dependencies on any injected services that are used inside a computed property. Even if such a dependency isn't always necessary in practice, it should be harmless for you to add it, and fortunately the rule can automatically add it for you.

@rwjblue @Turbo87 can you confirm?

**Comment by Turbo87** [comment]
I would agree with @BlueRaja. Since `myService` never changes it is probably unnecessary to add it as an explicit dependency. Unfortunately it's quite hard to statically determine the list of injected services for linting though. 🤔

**Comment by BlueRaja** [comment]
I know absolutely nothing about how eslint works, but wouldn't it be possible to look at the syntax tree to determine which properties are just calls to `@ember/service/inject`?  That would cover 99.9% of cases.

I know including services in the dependency list won't hurt anything, but _requiring_ the devs to do it in order to pass this check is a huge pain in the butt.  I ran the autofixer on our codebase - it made _hundreds_ of changes, and all but maybe 10 were superfluous.

**Comment by bmish** [comment]
Thanks for the feedback. It sounds like adding services as dependencies isn't strictly necessary (unless someone were to write `this.set('service', ...)` for some reason) and causes extra churn for those wanting to adopt this rule. As a result, I'll consider trying to update the rule to ignore service dependencies by default (with an option to catch services if desired).
